drivers/i2c/Kconfig

```diff
@@ -54,6 +54,19 @@ config I2C_CALLBACK
 	help
 	  API and implementations of i2c_transfer_cb.
 
+config I2C_ALLOW_NO_STOP_TRANSACTIONS
+	bool "Allows I2C transfers with no STOP on the last transaction [DEPRECATED]"
+	depends on !I2C_NRFX_TWI
+	depends on !I2C_NRFX_TWIM
+	depends on !I2C_STM32
+	depends on !I2C_GD32
+	depends on !I2C_ESP32
+	depends on !I2C_DW
```

Comment:
@fabiobaltieri shouldn't you
`select DEPRECATED` here?

Reply:
sure, added